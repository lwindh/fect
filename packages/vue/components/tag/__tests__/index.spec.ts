import { mount } from '@vue/test-utils'
import Tag from '..'

const Wrapper = {
  components: {
    [Tag.name]: Tag,
  },
  data() {
    return {
      invert: false,
      type: 'default',
      text: 'hello world',
    }
  },
  template: `
   <div class="container">
    <fe-tag :useInvert="invert" :type="type" :text="text" :type-data="type" />
   </div>
  `,
}

describe('Tag', () => {
  it('should render as a element', () => {
    const wrapper = mount(Tag)
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('component props should be work correctly', async () => {
    const wrapper = mount(Wrapper)
    await wrapper.setData({ type: 'success' })
    const el = wrapper.find('.fect-tag')
    expect(el.attributes('type-data')).toBe('success')
    await wrapper.setData({ invert: true })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
