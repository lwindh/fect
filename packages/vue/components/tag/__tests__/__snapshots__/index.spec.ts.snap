// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component props should be work correctly 1`] = `
<div class="container">
  <div class="fect-tag fect-tag--success fect-tag--invert" type-data="success">hello world</div>
</div>
`;
